cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -362,6 +372,14 @@ boolean scanApplications(ScanApplicationsRequest request, int txBatchSize,
    */
   Collection<ApplicationSpecification> getAllAppVersions(ApplicationId id);
 
+  /**
+   * Returns the latest version of an application
+   *
+   * @param id application id
+   * @return The metadata information of the latest application version.
+   */
+  ApplicationMeta getLatestAppVersion(ApplicationId id);
```

Comment:
I think `getLatestApp` is a better name since this returns the app meta, not a version